feat(ci): don't mount /var/cache/nixery from tmpfs into docker container

flokli opened the pull request:
With https://github.com/google/nixery/pull/127, nixery will use extended
attributes to store metadata (when using local storage).

Right now, our integration test mounts a tmpfs to /var/cache/nixery.
However, *user* xattrs aren't supported with tmpfs [1], so setting
xattrs would fail.

To workaround this, use a folder in the current working directory and
hope it's backed by something supporting user xattrs (which is the case
for GitHub Actions).

[1]: https://man7.org/linux/man-pages/man5/tmpfs.5.html#NOTES

flokli commented:
This should be merged before https://github.com/google/nixery/pull/127.

tazjin commented on .gitignore:
```diff
@@ -7,3 +7,6 @@ debug/
 *.p12
 *.json
+
+# Created by the integration test, and we want to keep things clean.
```
I think the stuff after the comma is redundant

flokli replied:
Removed.

tazjin commented on scripts/integration-test.sh:
```diff
@@ -10,9 +10,14 @@ echo "Loaded Nixery image as ${IMG}"
-docker run --rm -p 8080:8080 --name nixery \
+# We can't just mount a tmpfs to /var/cache/nixery, as tmpfs doesn't support
+# user xattrs.
+# So create a temporary directory in the current working directory, and hope
```
Instead of hoping, is there some way to check this before running the rest of the rest and then fail with a descriptive error?

flokli replied:
We could try setting some xattrs in there. However, that's exactly what #127 will do too.

flokli replied:
I added another line, that nixery will complain about unable to set xattrs if the underlying storage doesn't support it.

tazjin commented on scripts/integration-test.sh:
```diff
@@ -10,9 +10,14 @@ echo "Loaded Nixery image as ${IMG}"
+docker run --privileged --rm -p 8080:8080 --name nixery \
   -e PORT=8080 \
-  --mount type=tmpfs,destination=/var/cache/nixery \
+  --mount type=bind,source="$(pwd)"/var-cache-nixery,target=/var/cache/nixery \
```
I'd probably quote the whole parameter and interpolate ${PWD} instead, but it's not important.

flokli replied:
Updated, as I was at it anyways.

flokli commented:
I updated the PR to include your proposed changes - and also added some code removing `var-cache-nixery` if it already exists (leaving it around after the run for inspection). It doesn't matter much in CI, but for interactive tinkering.
